ctrd/container_types.go

```diff
@@ -11,12 +11,13 @@ import (
 // then create container by specifying the snapshot;
 // The other is create container by specify container rootfs, we use `RootFSProvided` flag to mark it,
 type Container struct {
-	ID      string
-	Image   string
-	Runtime string
-	Labels  map[string]string
-	IO      *containerio.IO
-	Spec    *specs.Spec
+	ID          string
+	Image       string
+	Runtime     string
+	Labels      map[string]string
+	IO          *containerio.IO
+	Spec        *specs.Spec
+	SnapshotKey string
```

Comment:
What is this used for?

Reply:
`SnapshotKey` is the snapshot id used by container.

we used to use container id as snapshot id, but when we support `upgrade ` a container, the snapshot id may be different with container id.

So, we must record the snapshot id that container used